qa/zcash/updatecheck.py

```diff
@@ -112,24 +112,22 @@ def get_dependency_list():
 
     return dependencies
 
-class GitHubToken:
-    def __init__(self):
-        token_path = os.path.join(SOURCE_ROOT, ".updatecheck-token")
-        try:
-            with open(token_path, encoding='utf8') as f:
-                token = f.read().strip()
-                self._user = token.split(":")[0]
-                self._password = token.split(":")[1]
-        except:
-            print("Please make sure a GitHub API token is in .updatecheck-token in the root of this repository.")
-            print("The format is username:hex-token.")
-            sys.exit(1)
-
-    def user(self):
-        return self.user
-
-    def password(self):
-        return self.password
+def parse_token():
+    token_path = os.path.realpath(os.path.join(SOURCE_ROOT, ".updatecheck-token"))
+    if not os.path.exists(token_path):
+        token_path = os.path.join(xdg.xdg_data_home(), "zcash/updatecheck/token")
```

Comment:
You can avoid the `xdg` dependency by using:
```suggestion
        token_path = os.path.expanduser("~/zcash/updatecheck/token")
```
(This works on all platforms.)